Add documentation for MsgSlashValidator (#152)
dydxprotocol/v4-documentation: pages/governance/slashing_a_validator.md

- The x/staking `HistoricalInfo` query endpoint can be used to find the correct value for `tokens_at_infraction_height`.
- See the [MsgSlashValidator](https://github.com/dydxprotocol/v4-chain/blob/protocol/v4.0.0/proto/dydxprotocol/govplus/tx.proto#L19) inline comments for further details on the above requirements.

### Example Proposal Json

Below is an example proposal JSON file to propose a slashing a validator's total bonded tokens (both user delegated and self-delegated) by 0.2 at block height 5000. In other words:
* assuming that the proposal passes governance vote
* assuming that the validator has 1000 total bonded tokens at height 5000
* once the proposal is passed, the validator will lose 200 tokens
* if there are unbonding undelegations and redelegations since height 5000, they will lose 0.2 of their stake, and the remaining will be taken from the validator for a total of 200 tokens lost
```json
{
    "title": "Slash a validator",
    "summary": "We are proposing to slash this misbehaving validator for X reasons.",
    "deposit": "10000000000000000000000adv4tnt",
    "messages": [
        {
          "@type": "/dydxprotocol.govplus.MsgSlashValidator",

>           "authority": "dydx10d07y265gmmuvt4z0w9aw880jnsr700jnmapky",

Reviewer: would be useful to add a comment that this has to be `gov` module account

# Slashing a validator

The chain supports slashing of misbehaving validators through governance vote.

## Proposal Message

The proposal should contain a single [MsgSlashValiator](https://github.com/dydxprotocol/v4-chain/blob/protocol/v4.0.0/proto/dydxprotocol/govplus/tx.proto#L19) message for each validator that should be slashed. See the inline comments for details on each field.

> **💡 Important: Both `tokens_at_infraction_height` and `slash_factor` must be set correctly, otherwise undelegations and redelegations might be slashed disproportionately to the rest of the validator's stake.**

Reviewer: Can we add explanations around how to pick these values? For example, how do you pick the right value for the height given the governance process takes a few days and etc

### Example Proposal Json

> Below is an example proposal JSON file to propose a slashing a validator's stake by 0.2. After execution the validator will lose 20% of their `tokens_at_infraction_height`. 

Reviewer: I think we should be specific about "validator's stake"

```suggestion
Below is an example proposal JSON file to propose a slashing a validator's total bonded tokens (= total stake, which includes both user delegated and self-delegated) by 0.2 at block height 5000. In other words:
* assuming that the proposal passes governance vote
* assuming that the validator has 1000 total bonded tokens at height 5000
* once the block height reaches 5000, the validator will lose 200 tokens
```

```json
{
    "title": "Slash a validator",
    "summary": "We are proposing to slash this misbehaving validator for X reasons.",
    "deposit": "10000000000000000000000adv4tnt",
    "messages": [
        {
          "@type": "/dydxprotocol.govplus.MsgSlashValidator",
          "authority": "dydx10d07y265gmmuvt4z0w9aw880jnsr700jnmapky",
          "validator_address": "dydxvalcons1z79h40nmd777scs93qjxaeak8m2cl6hpqg2rx9",
          "slash_factor": "0.2",

>           "tokens_at_infraction_height": "500000000000000000000000",

Reviewer: To make it easier to reference in other places

```suggestion
          "tokens_at_infraction_height": "5000",
```

          "infraction_height": 7647362
        }
    ]
}
  ```

## Submitting an Onchain Proposal

Follow instructions [here](./submitting_a_proposal.md) to submit an onchain proposal.

#### Disclaimer

> Users considering using the permissionless markets function of the dYdx v4 software are 

Reviewer: ```suggestion
Users considering using the permissionless markets function of the dYdX v4 software are 
```

Reviewer 2: Why is this paragraph relevant for this page?
